.github/workflows/tf-github-actions-oidc-provider.yml

```
@@ -0,0 +1,51 @@
+name: tf-github-actions-oidc-provider
+run-name: tf-github-actions-oidc-provider ${{ (inputs.apply || (github.event_name == 'push' && github.ref == 'refs/heads/main')) && 'apply' || 'plan' }}
+
+on:
+  push:
+    paths:
+      - .github/workflows/tf-github-actions-oidc-provider.yml
+      - terraform/services/github-actions-oidc-provider/**
+  workflow_dispatch:
+    inputs:
+      apply:
+        required: false
+        type: boolean
+        description: "Apply the terraform?"
+
+jobs:
+  check-fmt:
+    runs-on: codebuild-cdap-${{github.run_id}}-${{github.run_attempt}}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./actions/setup-tfenv-terraform
+      - run: terraform fmt -check -diff -recursive terraform/services/github-actions-oidc-provider
+
+  plan-apply:
+    needs: check-fmt
+    permissions:
+      contents: read
+      id-token: write
+    runs-on: codebuild-cdap-${{github.run_id}}-${{github.run_attempt}}
+    defaults:
+      run:
+        working-directory: ./terraform/services/github-actions-oidc-provider
+    strategy:
+      fail-fast: false
+      matrix:
+        app: [ab2d, bcda, dpc]
+        env: [dev, test, sandbox, prod]
+        include:
+          - app: cdap
+            env: mgmt
```

This resource is only needed once in each account.

```suggestion
        app: [bcda]
        env: [test, prod]
```